### PR TITLE
Remove + swap `PLANNER_XY_FEEDRATE_MM_S` w/ `XY_PROBE_FEEDRATE_MM_S`

### PR DESCRIPTION
**Pull request by classicrocker883**
### Description
This **PR** removes and replaces  `PLANNER_XY_FEEDRATE_MM_S` w/ `XY_PROBE_FEEDRATE_MM_S`

> [!IMPORTANT]
> In summary, with certain configurations, some moves are defined as `max_feedrate_mm_s` per `PLANNER_XY_FEEDRATE_MM_S`.
> This can have unintentional consequences using the **Max Feed rate** for simple moves.
> So rather than that, it uses **Default Homing Feed rate** for these Planner moves.

Something that stood out to me is `planner.settings.max_feedrate_mm_s[*_AXIS]`, which is defined by `DEFAULT_MAX_FEEDRATE`, which **X/Y_AXIS** is usually something in the hundreds.

However, in **motion.h**, we have:
```y
#if HAS_ABL_NOT_UBL
  extern feedRate_t xy_probe_feedrate_mm_s;
  #define XY_PROBE_FEEDRATE_MM_S xy_probe_feedrate_mm_s
#elif defined(XY_PROBE_FEEDRATE)                    // has a probe
  #define XY_PROBE_FEEDRATE_MM_S MMM_TO_MMS(XY_PROBE_FEEDRATE)
#else                                               // has No probe
  #define XY_PROBE_FEEDRATE_MM_S PLANNER_XY_FEEDRATE_MM_S
#endif
```

so when *not* using a probe it is `#define XY_PROBE_FEEDRATE_MM_S PLANNER_XY_FEEDRATE_MM_S`  
and `PLANNER_XY_FEEDRATE_MM_S` can be: 
(from **planner.h**)
```y
#if HAS_Y_AXIS
  #define PLANNER_XY_FEEDRATE_MM_S _MIN(planner.settings.max_feedrate_mm_s[X_AXIS], planner.settings.max_feedrate_mm_s[Y_AXIS])
#elif HAS_X_AXIS
  #define PLANNER_XY_FEEDRATE_MM_S planner.settings.max_feedrate_mm_s[X_AXIS]
#else
  #define PLANNER_XY_FEEDRATE_MM_S 60.0f
#endif
```
so basically, `PLANNER_XY_FEEDRATE_MM_S` should be something like "60", but being defined as `max_feedrate_mm_s` in the hundreds doesn't make sense. so `XY_PROBE_FEEDRATE_MM_S = PLANNER_XY_FEEDRATE_MM_S = max_feedrate_mm_s` doesn't make sense.

that is why `XY_PROBE_FEEDRATE_MM_S` should be `XY_PROBE_FEEDRATE`, which is `((homing_feedrate_mm_m.x + homing_feedrate_mm_m.y) / 2)`: as defined by `homing_feedrate_mm_m` = `HOMING_FEEDRATE_MM_M` = `{ (50*60), (50*60), (4*60) }`

### CODE REVIEW

File: Marlin/src/gcode/bedlevel/G26.cpp [hunk 1]
```diff
@@ -131,11 +131,11 @@
 #endif
 
 #ifndef G26_XY_FEEDRATE
-  #define G26_XY_FEEDRATE (PLANNER_XY_FEEDRATE_MM_S / 3.0)
+  #define G26_XY_FEEDRATE (XY_PROBE_FEEDRATE_MM_S / 3.0)
 #endif
 
 #ifndef G26_XY_FEEDRATE_TRAVEL
-  #define G26_XY_FEEDRATE_TRAVEL (PLANNER_XY_FEEDRATE_MM_S / 1.5)
+  #define G26_XY_FEEDRATE_TRAVEL (XY_PROBE_FEEDRATE_MM_S / 1.5)
```
Comment (thinkyhead) [inline Marlin/src/gcode/bedlevel/G26.cpp:138]: G26 is not a probing procedure, and "probing" feedrate is not meant to apply to non-probing situations. It is in fact meant to provide a reduced feedrate for situations where this benefits the probe type.

There may be better ways to define `PLANNER_XY_FEEDRATE_MM_S` based on user settings so that it is more in the range of a feedrate that is blessed for travel moves, but it is not appropriate to remove this conditional value altogether. The number "60.0" was not chosen because it is a typical maximum feedrate, but because 60 was the original "safe" default, a not-too-slow speed that pretty much any printer can achieve.

The max feedrate numbers are intended to be set to the fastest speed your printer can achieve without losing position, binding, grinding, etc., and your travel acceleration speed is intended to ensure that it can handle the feedrate request when starting from a cold stop. It's important to use these settings appropriately and make sure the max feedrate is not set beyond what the machine can manage. Most example configs have this value set pretty conservatively.

Another available value, when it exists, is `manual_feedrate_mm_m` in `planner.h` based on setting `MANUAL_FEEDRATE`. This applies to moves made with the LCD controller or host, of course, but it will fall in the same range as travel feedrate. The planner could choose to pick that up when it exists instead of the max axis feedrates.
